vulcanai/models/basenetwork.py

```diff
@@ -665,14 +665,16 @@ def fit(self, train_loader, val_loader, epochs,
                     self.lr_scheduler.step(epoch=epoch)
 
                 if self.early_stopping:
-                    if self.early_stopping_metric == "loss":
-                        if valid_loss:
+
+                    # only call if you've actually measured these.
+                    if valid_loss:
+                        if self.early_stopping_metric == "loss":
                             valid_loss_neg = -1 * valid_loss
-                        early_stopping(valid_loss_neg, self)
-                    elif self.early_stopping_metric == "accuracy":
-                        early_stopping(valid_acc, self)
-                    else:
-                        raise ValueError("Not a valid stopping metric")
+                            early_stopping(valid_loss_neg, self)
+                        elif self.early_stopping_metric == "accuracy":
+                            early_stopping(valid_acc, self)
```

Comment:
you only check if `valid_loss` is populated, perhaps you should also check if `valid_acc` also is populated?

Reply:
Yeah, redundant for now given the current structure but may as well add in case something changes